src/components/TokenAdministrative.js

```diff
@@ -59,10 +57,8 @@ class TokenAdministrative extends React.Component {
   }
 
   componentDidUpdate = (prevProps) => {
-    if (this.props.historyTransactions !== prevProps.historyTransactions) {
-      if (this.props.token.uid !== prevProps.token.uid) {
-        this.cleanStates();
-      }
+    if ((this.props.tokensHistory && !prevProps.tokensHistory) ||
+        (this.props.tokensHistory[this.props.token.uid].length !== prevProps.tokensHistory[this.props.token.uid].length)) {
```

Comment:
I feel we can avoid this whole method passing just the selected token history to the component.

Reply:
I changed a bit your suggestion but completely removed the `componentDidUpdate` method. Token history was not used in the component, it was only used on state to understand when a new tx for the token was received to re-render the page. I used as key of component a concatenation of `tokenUid-tokenHistoryLen`. With this, every time a new tx for this token arrives the component will re-render. What do you think?

Reply:
It seems a clever idea. I've never seen the key being used this way but it should work nicely. I like it.